astropy/units/tests/test_equivalencies.py

```diff
@@ -215,6 +217,61 @@ def test_massenergy():
                        pow_eV.value, rtol=1e-7)
 
 
+@pytest.mark.parametrize(
+    'mass,length,time',
+    [[2 * u.solMass, 3 * u.km, 10 * u.microsecond],
+     [1 * u.kg, 7.54e-31 * u.km, 2.47e-36 * u.s],]
+)
+def test_mass_length_time_geometrized_equivalency(mass, length, time):
+    """Known cases of mass, length, time equivalencies. An example being
+    Schwarzschild radius for sun is 3 km, which is about 10 microsecond
+    of light crossing time.
+    """
+    converted_mass = mass.to('kg', equivalencies=u.geometrized('mass')).value
+    converted_length = length.to('m', equivalencies=u.geometrized('length')).value
+    converted_time = time.to('s', equivalencies=u.geometrized('time')).value
+
+    assert mass.to('kg').value == pytest.approx(converted_mass)
+    assert length.to('m').value == pytest.approx(converted_length)
+    assert time.to('s').value == pytest.approx(converted_time)
+
+
+@pytest.mark.parametrize(
+    'si_qty,geometrized_qty,physical_type',
+    [[1 * u.lightsecond**2, 1 * u.s**2, 'area'],
+     [1 * u.lightsecond**3, 1 * u.s**3, 'volume'],]
+)
+def test_area_volume_geometric_equivalency(si_qty, geometrized_qty,
+                                           physical_type):
+    """Geometric units of area and volume"""
+    converted_si_qty = geometrized_qty.to(
+        si_qty.unit,
+        equivalencies=u.geometrized(physical_type)
+    )
+    assert converted_si_qty == si_qty
+
+
+def test_mass_density_geometric_equivalency():
```

Comment:
Hi @mhvk based on the discussion we had earlier, I have changed the PR to include a few more physical types. I will mention in brief what all has been taken into account.
- Irreducible units mass, length, time can be converted into each other
- Composite units which are homogeneous like area i.e. length squared can be converted to mass squared etc
- The challenge is for heterogeneous composite units like density. Currently it works for kg/m^3 to 1/kg^2, 1/s^2 etc and the other which way. But not for kg/m^3 to g/s^3, or to 1/(m*s) etc.
Having support for all of these is a nice to have but will need more work. But let me know what you think of this until now. I have tried adding tests for each of the conversion, but if something is missing let me know. Adding support for other physical types like pressure/force is not difficult if we restrict the support to only target unit with a single base - like pressure in SI  to 1/(length or time or distance)**2, similar to mass density.

Also since the code has changed a lot since my initial PR, I am happy to rebase to a single commit if that helps. I have not yet taken care of the definition of the G = 1 vs. 8 pi G = 1, but its on my radar. Thanks